### PR TITLE
gh-125665: Update turtledemo docstrings with correct file names

### PR DESCRIPTION
**Pull request by Wulian233**
edit: skip news I think

* Issue: gh-125665

----
📚 Documentation preview 📚: https://cpython-previews--125691.org.readthedocs.build/

### CODE REVIEW

**Comment by terryjreedy** [comment]
About the side issue of sorting_animate.py:  The normal way to stop within the demo framework is the red STOP button, and this works (but see below).  The spacebar bit is for when one runs sorting_animate directly, as with `f:\dev\3x>python -m turtledemo.sorting_animate`.  Spacebar *does* work then.  Perhaps other demos should stop with spacebar when run directly.  Maybe some do?  Perhaps demo should know if run directly or in framework.  Anyway, another issue.  The 'but' above is that both STOP and spacebar raise Terminator when pressed while the animation is going, as opposed to when waiting for input (`i`, etc.).

File: Lib/turtledemo/colormixer.py
Comment (terryjreedy) [inline Lib/turtledemo/colormixer.py]: I would like all of these 3-line headers condensed to 

```suggestion
"""turtledemo/colormixer.py
```
This directory is the turtledemo suite and saying this twice is redundant.

File: Lib/turtledemo/sorting_animate.py
Comment (terryjreedy) [inline Lib/turtledemo/sorting_animate.py]: Let's leave code changes to a separate PR.  See https://github.com/python/cpython/pull/125691#issuecomment-2425136335 for this issue.

File: Lib/turtledemo/chaos.py
Comment (terryjreedy) [inline Lib/turtledemo/chaos.py]: I think remove this.  Each example was written by somebody.

File: Doc/library/turtle.rst
Comment (terryjreedy) [inline Doc/library/turtle.rst]: I think this turtledemo item can be deleted.  Both examples are listed in the table above.

The 2.6 section should be deleted and I doubt this section is current.  'Added' notes should follow the item added, not collected together like this.  But these are turtle doc, not a turtledemo concern.

File: Lib/turtledemo/forest.py
Comment (terryjreedy) [inline Lib/turtledemo/forest.py]: Remove this link (German).  It has nothing to do with with the Logo program.  Anyone wanting to know about him should do a normal search.

**Review by terryjreedy — CHANGES_REQUESTED** [review]

**Comment by terryjreedy** [comment]
As explained in https://github.com/python/cpython/issues/118673#issuecomment-2419980443, backports will have merge conflicts, like #123457 did.  So fixing more than the 3 header lines now is likely a good idea.  I believe that if you have the diff in a file, you can just add `#!/usr/bin/env python3\n` above `""".*suite:\n` and `"""turtledemo.*\n` lines (this latter after my requested change).  Then apply in 3.13 PR branch.  But I don't currently know all the details (I'm not a git expert).  We would have to check all examples in 3.13 to make sure they have the same exact `#!` line.

**Comment by terryjreedy** [comment]
I decided to skip news as there is no reason I can think of that anyone should be hinted to possibly read the revised docstring.

**Comment by Wulian233** [comment]
> As explained in [#118673 (comment)](https://github.com/python/cpython/issues/118673#issuecomment-2419980443), backports will have merge conflicts, like #123457 did. So fixing more than the 3 header lines now is likely a good idea. I believe that if you have the diff in a file, you can just add above and lines (this latter after my requested change). Then apply in 3.13 PR branch. But I don't currently know all the details (I'm not a git expert).`#!/usr/bin/env python3\n``""".*suite:\n``"""turtledemo.*\n`
> 
> EDIT: Cancel this: "We would have to check all examples in 3.13 to make sure they have the same exact line. " As far as I can see, the turtledemo lines removed from main by https://github.com/python/cpython/pull/119658/files are all the same. Some non-turtledemo files had a space after .`#!``#!`

Sorry I didn't understand what should I do :( . Do I need to open another PR of 3.13/3.12 to resolve the conflict after merge? The above suggestion has been resolved, thanks

File: Lib/turtledemo/colormixer.py
Comment (terryjreedy) [reply Lib/turtledemo/colormixer.py]: This is not resolved because it was done only for colormixer, not all files.  Please let me (and other coredevs) decide if our comments are resolved.

File: Lib/turtledemo/sorting_animate.py
Comment (terryjreedy) [reply Lib/turtledemo/sorting_animate.py]: Code changes are moved to #125763 and its PR.

File: Doc/library/turtle.rst
Comment (terryjreedy) [reply Doc/library/turtle.rst]: The two lines were deleted.

File: Lib/turtledemo/forest.py
Comment (terryjreedy) [reply Lib/turtledemo/forest.py]: I meant to only remove the quoted weblink, not the credit line "The example ... written by Erich Neuwirth."  Please restore it.

**Comment by terryjreedy** [comment]
I unresolved comments 1 and 5 (though gh falsely still claims outdated).  Fix 5 first.

I believe 1 can be done with sed, which comes with Git for Windows.  Start menu => Git => Git Bash starts a Linux-like terminal that recognizes sed (stream editor) commands.  I used it several years ago after finding and reading a sed manual.  I will do so if need be.  (It might be easier to first undo the change to clock.py in the local branch version -- I don't know if sed on multiple files stops on error or merely reports and continues with the next one.)

When I merge to main, I will try backporting to be sure that the merge conflict for clock.py was due to (or remains after) changing the docstring rather than only resulting from deleting the coding line.  Assuming the backport fails, the first step will be to switch to a 3.13 branch (or workspace) and copy the new main turtledemo files into the turtledemo directory of a 3.13 PR branch.  I may ask other core developers if we need to add back the #! line.  If so, sed would be easiest.  I will let you try if you wish or refresh my sed-fu.

File: Lib/turtledemo/colormixer.py
Comment (Wulian233) [reply Lib/turtledemo/colormixer.py]: Thanks for informing, I'll be aware of that in the future

**Comment by terryjreedy** [comment]
Looks good as of my last comments.  Tomorrow I will recheck.  Since there is likely to be a backport problem, I will also expand comments to see if I want to make any further changes now before merging.

File: Lib/turtledemo/chaos.py
Comment (terryjreedy) [inline Lib/turtledemo/chaos.py]:
```suggestion
A demonstration of chaos.
```

File: Lib/turtledemo/forest.py
Comment (terryjreedy) [reply Lib/turtledemo/forest.py]: Done.

**Comment by terryjreedy** [comment]
I added periods to 3 files.  Any further changes past line 3 will backport properly.

**Review by terryjreedy — APPROVED** [review]

**Comment by terryjreedy** [comment]
Of the 19 examples, 13 have shebang lines an have the executable mode bit; the other 6 have neither.  Using cherry_picker showed that the 13 have merge conflicts because of the shebang lines.  I think we should leave shebang line presence and file mode alone.  I am thinking about how to be best do this while changing the docstring.

**Comment by terryjreedy** [comment]
Thank you for the main PR. I can think of 3 backport approaches, and all seem more trouble than the result is worth.  I am now inclined to skip backports, close the issue, and move on to other isssues.

**Comment by Wulian233** [comment]
If you'd like, I can create a manual target of 3.13  `[3.13] 
gh-xxx: `, which may also resolve potential conflicts in the future (I'm not sure if the bug with the space exit https://github.com/python/cpython/issues/125763 after the fix has any conflicts)

**Comment by terryjreedy** [comment]
At least part of the reason that shebang/mode changes were not backported was compatibility considerations (https://peps.python.org/pep-0387/). ([IDLE exception](https://peps.python.org/pep-0434/). ) As I said above, "we should leave shebang line presence and file mode alone".  This multiplies the time needed for a [3.13] backport by at least 2.  If you produce one ready-to-merge for fun/instruction, I will merge it, but I prefer to work on other issues.

**Pull request merged**
